clang/include/clang/Tooling/DependencyScanning/ModuleDepCollector.h

```diff
@@ -181,7 +181,8 @@ struct ModuleDeps {
   /// including transitive dependencies.
   std::vector<std::string> FileDeps;
 
-  std::variant<std::monostate, CowCompilerInvocation, std::vector<std::string>>
+  mutable std::variant<std::monostate, CowCompilerInvocation,
```

Comment:
I have this change included here to provide context,  but it's in a separate commit to be able to port to upstream, if we agree on this direction.

Reply:
https://github.com/llvm/llvm-project/pull/132968